server/src/main/java/org/opensearch/tasks/ResourceStatsInfo.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tasks;
+
+/**
+ *  Defines resource stats information.
+ */
+public class ResourceStatsInfo {
+    private final long startValue;
+    private long endValue;
```

Comment:
Shouldn't we have `volatile` here? and also for other fields holding data which is being updated and referenced by different threads.

Reply:
endValue is updated to AtomicBoolean now.